### PR TITLE
Oth: User documentation on the inner workings of workload migration

### PR DESCRIPTION
**Pull request by jistr**
Also i'm piggy-backing a Dev commit into the pull request re tagging data seed tasks in tests, which came in handy for demo preparation.

### CODE REVIEW

**Review by philroche — APPROVED** [review]
Excellent and nice diagrams too

**Review by ccamacho — CHANGES_REQUESTED** [review]

File: docs/src/user/how-it-works-workload-migration.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+The workload migration utilizes so called *conversion hosts* to
+transfer data from the source cloud to the destination cloud.
+
+The conversion hosts are temporarily deployed in the source &
+destination projects. Being in the same project as the source (and
+destination, respectively) VMs ensures that the conversion hosts will
+have access to the data that needs to be migrated (snapshots and
+volumes).
+
+.. figure:: https://raw.githubusercontent.com/os-migrate/os-migrate/main/media/walkthrough/2020-06-24-osp-migrate-fig5.png?raw=true
```
[the remark this answers is not in the document]
Comment (jistr) [reply docs/src/user/how-it-works-workload-migration.rst:21]: Yes but i'd perhaps do another PR with this? I didn't want to refactor everything until we get the rendering mechanism in place.

File: Makefile [hunk 2]
```diff
@@ -150,9 +150,12 @@ test-unit: reinstall
 
 # DOCS
 
-docs: reinstall
+docs: reinstall docs-diagrams
 	./scripts/docs-build.sh
 
+docs-diagrams:
+	plantuml -tsvg -output render ./docs/src/images/plantuml/*.plantuml
```
[the remark this answers is not in the document]
Comment (ccamacho) [reply Makefile:157]: ++

[the remark this answers is not in the document]
Comment (jistr) [reply Makefile]: Done, using DPI 150 for rendering which seems to be alright for the web. Also rebased onto main branch as i wanted both toolbox edits at the same time (using Fedora 34 && the new packages for plantuml).

**Review by ccamacho — APPROVED** [review]

**Pull request merged**